src/mlpack/core/data/split_data.hpp

```diff
@@ -47,13 +48,15 @@ namespace data {
  * @param shuffleData If true, the sample order is shuffled; otherwise, each
  *       sample is visited in linear order. (Default true.)
  */
```

Comment:
It would probably be good to document the new label type template parameter here (and what types it's supposed to take).

Reply:
Agreed

Reply:
No problem, although sometimes I think it can be easier to write the documentation first (to decide on the interface), and *then* write the implementation.  (What you're doing here is just fine, I just wanted to write down the thought. :))